notification_service_tests.sh

```diff
@@ -69,8 +77,29 @@ function init_db() {
     rm config.toml
 }
 
+# mechanism to chain more trap commands added in different parts of this script
+exit_trap_command=""
+function cleanup {
+    eval "$exit_trap_command"
+}
+trap cleanup EXIT
+
+function add_exit_trap {
```

Comment:
oh, I thought you can use as many `trap` as you want. I guess just the last one is preserved and this is why you had to create this function?

Reply:
Yes, exactly. With each call of trap for given signal, previous command is overwritten with new one. So without this, a lot of uvicorn mock services stayed running after tests finished.

Reply:
True!